src/julia-syntax.scm

```diff
@@ -2478,32 +2478,55 @@ end
 f(x) = yt(x)
 |#
 
-;; template for generating a closure type with parameters
-(define (type-for-closure-parameterized name P fields types super)
-  (let ((n (length P)))
-    `(thunk
+(define (type-for-closure-parameterized name P names fields types super)
+   (let ((n (length P)))
+    `((thunk
```

Comment:
It also doesn't need to return a lambda, just a couple of statements, but lowering handled that wrong